packages/react-core/src/components/DataList/DataListAction.tsx

```diff
@@ -27,44 +27,20 @@ export interface DataListActionProps extends Omit<React.HTMLProps<HTMLDivElement
   isPlainButtonAction?: boolean;
 }
 
-interface DataListActionState {
-  isOpen: boolean;
-}
-
-export class DataListAction extends React.Component<DataListActionProps, DataListActionState> {
-  static displayName = 'DataListAction';
-  state = {
-    isOpen: false
-  };
-
-  onToggle = (isOpen: boolean) => {
-    this.setState({ isOpen });
-  };
-
-  onSelect = () => {
-    this.setState(prevState => ({
-      isOpen: !prevState.isOpen
-    }));
-  };
-
-  render() {
-    const {
-      children,
-      className,
-      visibility,
-      /* eslint-disable @typescript-eslint/no-unused-vars */
-      id,
-      'aria-label': ariaLabel,
-      'aria-labelledby': ariaLabelledBy,
-      isPlainButtonAction,
-      /* eslint-enable @typescript-eslint/no-unused-vars */
-      ...props
-    } = this.props;
-
-    return (
-      <div className={css(styles.dataListItemAction, formatBreakpointMods(visibility, styles), className)} {...props}>
-        {isPlainButtonAction ? <div className={css(styles.dataListAction)}>{children}</div> : children}
-      </div>
-    );
-  }
-}
+export const DataListAction: React.FunctionComponent<DataListActionProps> = ({
+  children,
+  className,
+  visibility,
+  /* eslint-disable @typescript-eslint/no-unused-vars */
+  id,
+  'aria-label': ariaLabel,
+  'aria-labelledby': ariaLabelledBy,
+  isPlainButtonAction,
```

Comment:
Would be great to update this section. The isPlainButtonAction prop is used and the other three either don't have to be destructured or should be explicitly mapped in the div

Reply:
sure, I was just wondering if these aria-labels are desired there or should be instead deprecated and not used as properties at all.
Because I think it's common to use aria-label with div element but then they need role as well.

Reply:
So I initially implemented to pass these 'props' but the tests caught, that in the scenario, when we have 2 actions for the same item which appear conditionally on the UI according to the screen size, then we end up with duplicate Ids, which is html dom validity violation.

Then we could of course provide different 'id's per action, but that would make card the 'aria-labelled-by' asignment.

Anyway, let's take a look into that into another PR. @redallen